CMakeLists.txt

```diff
@@ -239,6 +240,45 @@ endif()
 
 include(cmake/optional.cmake)
 
+set(hardened_cxx_flags "")
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  # All versions of gcc that we commonly use for building are subject to bug
```

Comment:
This comment is outdated compared to master.

Reply:
Implemented approach from https://github.com/bitcoin/bitcoin/pull/28672,